mtps-get-task

```diff
@@ -657,60 +712,75 @@ buildtags="$(buildtags_from_answer "$listbuildtags")"
-        srpm_pkg_name="$(from_rpm_name "${srpm_pkg_file}" "name")"
-        if [ -z "$nvr" ]; then
-            nvr="${srpm_pkg_file%%.src.rpm}"
-        fi
-        build="$(brew_get_build "$nvr")"
-        volume_name="$(build2volume_name "$build")"
-        build_source="$(build2build_source "$build")"
-        srpm_url="$(mk_url "$srpm" "$srpm_pkg_file" "$is_scratch")"
+task_method="$(taskinfo2method "$task_info")"
+debug "task_method: $task_method"
+if [[ "$task_method" == "build" ]]; then
+    TASKS="$TASK"
+    if [ -n "$RECURSIVE" ]; then
+        task_children="$(brew_get_task_children "$TASK")"
+        task_children="$(children_from_answer "$task_children")"
+        TASKS="$TASKS $task_children"
+        TASKS="${TASKS%% }"
     fi
-    rpms_from_task_all+=" $rpms_from_task"
-    rpms_from_task_noarch_all+=" $rpms_from_task_noarch"
-done
 
-# remove trailing whitespaces
-rpms_from_task_all="${rpms_from_task_all%%+( )}"
-rpms_from_task_noarch_all="${rpms_from_task_noarch_all%%+( )}"
+    srpm_url=
+    volume_name=
+    build_source=
+    srpm_pkg_file=
+    srpm_pkg_name=
+    rpms_arch_all=""
+    rpms_noarch_all=""
+    for task in $TASKS; do
+        task_result="$(brew_get_task_result "$task")" # Holds plain XML
+        rpms_from_task="$(task_result2rpms "$ARCH" "$task_result")"
+        rpms_from_task_noarch="$(task_result2rpms "noarch" "$task_result")"
+        srpm="$(task_result2srpm "$task_result")"
+        if [ -n "$srpm" ]; then
+            srpm_pkg_file="$(basename "$srpm")"
+            srpm_pkg_name="$(from_rpm_name "${srpm_pkg_file}" "name")"
+            if [ -z "$nvr" ]; then
+                nvr="${srpm_pkg_file%%.src.rpm}"
+            fi
+            build="$(brew_get_build "$nvr")"
+            volume_name="$(build2volume_name "$build")"
+            build_source="$(build2build_source "$build")"
+            srpm_url="$(mk_url "$srpm" "$srpm_pkg_file" "$is_scratch")"
+        fi
+        rpms_arch_all+=" $rpms_from_task"
+        rpms_noarch_all+=" $rpms_from_task_noarch"
+    done
 
-echo "NVR: $nvr"
-echo "BUILD ID: $buildid"
-echo "BUILD TAGS: $buildtags"
-echo "SRPM PKG NAME: $srpm_pkg_name"
-echo "SRPM PKG FILE: $srpm_pkg_file"
-echo "SRPM URL: $srpm_url"
-echo "BUILD SOURCE: $build_source"
+    # remove trailing whitespaces
+    rpms_arch_all="${rpms_arch_all%%+( )}"
+    rpms_noarch_all="${rpms_noarch_all%%+( )}"
 
-if [ -n "$SRPM" ]; then
-    # Print only SRPM information
-    exit 0
+    echo "NVR: $nvr"
+    echo "SRPM PKG NAME: $srpm_pkg_name"
+    echo "SRPM PKG FILE: $srpm_pkg_file"
+    echo "SRPM URL: $srpm_url"
+    echo "BUILD SOURCE: $build_source"
+
+    if [ -n "$SRPM" ]; then
+        # Print only SRPM information
+        exit 0
+    fi
+# Builds imported from Konflux don't have descendant/child tasks to take the (S)RPMs from.
+# TODO: This is much easier than taking the (S)RPMs from (sub)task(s).
+# TODO: Could we use this even for the 'build' method?
```

Comment:
I think the major PITA here is scratch builds, that are technically don't qualify as "builds" in Koji. They don't have any build id...